### DISCUSSION TITLE
Connect edge all the direction (Top, Bottom, Left, Right)

### DISCUSSION THREAD

**Discussion by Hadiasemi** [xyflow/xyflow#3215] [Q&A]
Is there a way to connect node to middle of edge connected between two nodes? Also, how can I connect to all side of node with edges? I try different way such sourcePosition, targetPostion but it only activate two side which I want to be all sides. I really appreciate if you can give some help.

**Comment by moklick** [comment]
we have different examples that should help you to get started:

- https://reactflow.dev/docs/examples/nodes/easy-connect/ 
- https://reactflow.dev/docs/examples/edges/simple-floating-edges/
- https://reactflow.dev/docs/examples/edges/floating-edges/

**Reply by Hadiasemi** [reply]
Thanks, Do you have any recommendation on how to make each node color respond to the API endpoint change background color?
